mealpy.py

```diff
@@ -1,126 +1,111 @@
-import requests
 import time
 
+import requests
 from apscheduler.schedulers.blocking import BlockingScheduler
 
 BASE_DOMAIN = 'secure.mealpal.com'
-BASE_URL = 'https://' + BASE_DOMAIN
-LOGIN_URL = BASE_URL + '/1/login'
-CITIES_URL = BASE_URL + '/1/functions/getCitiesWithNeighborhoods'
-MENU_URL = BASE_URL + '/api/v1/cities/%s/product_offerings/lunch/menu'
-RESERVATION_URL = BASE_URL + '/api/v2/reservations'
-KITCHEN_URL = BASE_URL + '/1/functions/checkKitchen3'
-
-LOGGED_IN_COOKIE = 'isLoggedIn'
+BASE_URL = f'https://{BASE_DOMAIN}'
 
 HEADERS = {
     'Host': BASE_DOMAIN,
     'Origin': BASE_URL,
-    'Referer': BASE_URL + '/login',
+    'Referer': f'{BASE_URL}/login',
     'Content-Type': 'application/json',
 }
 
 
-class MealPal(object):
+class MealPal():
 
     def __init__(self):
-        self.headers = HEADERS
         self.cookies = None
         self.cities = None
-        self.schedules = None
 
     def login(self, username, password):
         data = {'username': username, 'password': password}
-        r = requests.post(
-            LOGIN_URL, data=json.dumps(data), headers=self.headers)
-        self.cookies = r.cookies
-        self.cookies.set(LOGGED_IN_COOKIE, 'true', domain=BASE_URL)
-        return r.status_code
-
-    def get_cities(self):
-        if not self.cities:
-            r = requests.post(CITIES_URL, headers=self.headers)
-            self.cities = r.json()['result']
-        return self.cities
+        url = f'{BASE_URL}/1/login'
+        request = requests.post(url, data=json.dumps(data), headers=HEADERS)
+        self.cookies = request.cookies
+        self.cookies.set('isLoggedIn', 'true', domain=BASE_URL)
+        return request.status_code
+
+    @staticmethod
+    def get_cities():
+        url = f'{BASE_URL}/1/functions/getCitiesWithNeighborhoods'
+        request = requests.post(url, headers=HEADERS)
+        return request.json()['result']
 
     def get_city(self, city_name):
-        if not self.cities:
-            self.get_cities()
-        city = next((i for i in self.cities if i['name'] == city_name), None)
+        city = next((i for i in self.get_cities() if i['name'] == city_name), None)
         return city
 
-    def get_schedules(self, city_name, city_id=None):
-        if not city_id:
-            city_id = self.get_city(city_name)['objectId']
-        r = requests.get(
-            MENU_URL % city_id, headers=self.headers, cookies=self.cookies)
-        self.schedules = r.json()['schedules']
-        return self.schedules
-
-    def get_schedule_by_restaurant_name(
-            self, restaurant_name, city_name=None, city_id=None):
-        if not self.schedules:
-            self.get_schedules(city_name, city_id)
-        restaurant = next(i for i in self.schedules if i['restaurant']['name'] == restaurant_name)
+    def get_schedules(self, city_name):
+        city_id = self.get_city(city_name)['objectId']
+        url = f'{BASE_URL}/api/v1/cities/%s/product_offerings/lunch/menu'
```

Comment:
Can we add `f'{BASE_URL}/api/v1/cities/{city_id}/product_offerings/lunch/menu'` here as well?

Reply:
Oops, good catch.